synapse/appservice/api.py

```diff
@@ -120,7 +120,11 @@ async def query_user(self, service: "ApplicationService", user_id: str) -> bool:
 
         uri = service.url + ("/users/%s" % urllib.parse.quote(user_id))
         try:
-            response = await self.get_json(uri, {"access_token": service.hs_token})
+            response = await self.get_json(
+                uri,
+                {"access_token": service.hs_token},
+                headers={"Authorization": f"Bearer {service.hs_token}"},
```

Comment:
Footgun alert: the values of headers should be a list of strings, not strings themselves. See #14301.